src/get_model_metadata_impl.cpp

```diff
@@ -103,8 +103,8 @@ void GetModelMetadataImpl::convert(
         *input.mutable_name() = name;
         *input.mutable_tensor_shape() = tensorflow::TensorShapeProto();
 
-        for (auto dim : tensor->getEffectiveShape()) {
-            input.mutable_tensor_shape()->add_dim()->set_size(dim ? dim : -1);
+        for (auto dim : tensor->getShape_3()) {
+            input.mutable_tensor_shape()->add_dim()->set_size(dim.getAnyValue() ? dim.getAnyValue() : DYNAMIC_DIMENSION);
```

Comment:
Add here TODO - potentialy we should extract logic what value we want to get for what value of dim which depends on context.

Reply:
This is to be done in separate task anyway.